quickwit/quickwit-metastore/src/metastore/postgres/metastore.rs

```diff
@@ -399,6 +403,34 @@ impl MetastoreService for PostgresqlMetastore {
         Ok(response)
     }
 
+    async fn update_index(
+        &mut self,
+        request: UpdateIndexRequest,
+    ) -> MetastoreResult<IndexMetadataResponse> {
+        let retention_policy_opt = request.deserialize_retention_policy()?;
+        let search_settings = request.deserialize_search_settings()?;
+        let index_uid: IndexUid = request.index_uid().clone();
+        let updated_metadata = run_with_tx!(self.connection_pool, tx, {
+            mutate_index_metadata::<MetastoreError, _>(
+                tx,
+                index_uid,
+                |mut index_metadata: IndexMetadata| {
+                    if index_metadata.index_config.search_settings != search_settings
```

Comment:
Use the setters.

Reply:
what setters are you referring to? The setters created in this PR are for the `FileBackedIndex` entity only, and I don't see any setters on `IndexMetadata`.

Reply:
I thought the setters were `IndexMetadata`. My bad.